newIDE/app/src/ObjectEditor/Editors/SpriteEditor/PointsEditor/PointRow.js

```diff
@@ -6,96 +6,144 @@ import IconButton from '../../../../UI/IconButton';
 import Delete from '@material-ui/icons/Delete';
 import Edit from '@material-ui/icons/Edit';
 import TextField from '../../../../UI/TextField';
+import SemiControlledTextField from '../../../../UI/SemiControlledTextField';
 import styles from './styles';
 import ThemeConsumer from '../../../../UI/Theme/ThemeConsumer';
 import Text from '../../../../UI/Text';
+import { roundTo } from '../../../../Utils/Mathematics';
+
+const POINT_COORDINATE_PRECISION = 4;
 
 type Props = {|
   pointName: string,
-  nameError: boolean,
+  nameError?: boolean,
   onBlur?: (ev: any) => void,
-  onRemove?: (ev: any) => void,
-  onEdit?: (ev: any) => void,
+  onRemove?: ?(ev: any) => void,
+  onEdit?: ?(ev: any) => void,
+  onClick: (pointName: string) => void,
+  onMouseEnter: (pointName: ?string) => void,
+  onMouseLeave: (pointName: ?string) => void,
+  selected: boolean,
   pointX: number,
   pointY: number,
   onChangePointX: (value: number) => void,
   onChangePointY: (value: number) => void,
-  isAutomatic?: Boolean,
+  isAutomatic?: boolean,
 |};
 
-const PointRow = (props: Props) => (
-  <ThemeConsumer>
-    {muiTheme => (
-      <TableRow
-        style={{
-          backgroundColor: muiTheme.list.itemsBackgroundColor,
-        }}
-      >
-        <TableRowColumn style={styles.handleColumn}>
-          {/* <DragHandle /> Reordering point is not supported for now */}
-        </TableRowColumn>
-        <TableRowColumn>
-          <TextField
-            margin="none"
-            defaultValue={props.pointName || 'Unnamed point'}
-            id={props.pointName}
-            fullWidth
-            errorText={
-              props.nameError ? 'This name is already taken' : undefined
-            }
-            disabled={!props.onBlur}
-            onBlur={props.onBlur}
-          />
-        </TableRowColumn>
-        <TableRowColumn style={styles.coordinateColumn}>
-          {!props.isAutomatic ? (
+const PointRow = ({ onMouseLeave, pointX, pointY, ...props }: Props) => {
+  const onLeave = React.useCallback(
+    () => {
+      onMouseLeave(null);
+    },
+    [onMouseLeave]
+  );
+
+  return (
+    <ThemeConsumer>
+      {muiTheme => (
+        <TableRow
+          style={{
+            backgroundColor: props.selected
+              ? muiTheme.listItem.selectedBackgroundColor
+              : muiTheme.list.itemsBackgroundColor,
+          }}
+          onClick={() => props.onClick(props.pointName)}
+          onMouseEnter={() => props.onMouseEnter(props.pointName)}
```

Comment:
I wonder if we should prefer onPointerEnter to handle other things than a mouse

Reply:
Yes, good idea! I didn't think of that